src/ebonite/ext/flask/server.py

```diff
@@ -75,31 +75,32 @@ class FlaskConfig(Config):
     FlaskConfig.log_params()
 
 
+PREBUILD_PATH = current_module_path('prebuild')
+BASE_IMAGE_TEMPLATE = 'zyfraai/flask:{}'
+
+
 class FlaskServer(BaseHTTPServer):
     """
     Flask- and Flasgger-based :class:`.BaseHTTPServer` implementation
     """
 
     additional_sources = [
-        current_module_path('build_templates', 'flask-site-nginx.conf'),
-        current_module_path('build_templates', 'nginx.conf'),
-        current_module_path('build_templates', 'supervisord.conf'),
-        current_module_path('build_templates', 'uwsgi.ini')
+        current_module_path('build', 'app.py')  # replace stub in base image
     ]
 
     # TODO somehow make this depend on builder implementation?
     additional_options = {
-        'templates_dir': os.path.join(os.path.dirname(__file__), TEMPLATES_DIR),
-        'run_cmd': '["/usr/bin/supervisord"]'
+        'templates_dir': current_module_path('build'),
+        'base_image': lambda python_version: BASE_IMAGE_TEMPLATE.format(python_version),
+        'run_cmd': False,  # base image has already specified command
+        'prebuild_hook': lambda python_version: prebuild_image(PREBUILD_PATH, BASE_IMAGE_TEMPLATE, python_version)
```

Comment:
maybe move it to separate class field?

Reply:
Do you mean move `prebuild_hook` to class field? Of which class? Please provide a bit more details on that. I did it here as existing options here were already used for Docker builder configuration. Thus I though it was a good fit.

Reply:
i think i meant Server class, not sure now if this will work.
but actually, the idea behind this comment is more general.
i like this idea of separating build into 2 stages - server and model. even if we do not prebuilt servers, still it allows to reuse build cache for example.
so it is natural to make this more "legit", because right now it looks like we hack our way around our own design, using "holes" we left to provide users a way to customize stuff.
so this proposition was one step towards this "legitimization", but maybe it is better to use complex approach and redisign this altogether